### PR TITLE
[chore] add template for new issues for unmaintained components

### LINKED ISSUE #32485: Add template for unmaintained component issues
### Component(s)

_No response_

### Describe the issue you're reporting

To reduce maintenance cost on maintainers and provide consistency, it would be useful to have a template when we need to open an issue marking a component as unmaintained.

### PR DESCRIPTION
**Pull request by codeboten**
Fixes #32485

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      label: 
+      description: Please provide details 
+      value: |
+        Code owners for this component have explicitly stated they will no longer maintain the code
+        or have been unresponsive to issues/pull requests.
+
+        - [ ] Wait 6 weeks for the code owner to provide feedback, either here or in the linked issue
+        - [ ] Attempt to recruit new code owners, potentially from the community (cc <code owners>)
+        - [ ] On <date 6 weeks from now>, 6 weeks from now, the component will be labeled as unmaintained if the original code owner is still unresponsive and no other code owner for the component has been found
+        - [ ] On <date 6 monnths from now>, 6 months from now, the component will be removed
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml:287]:
```suggestion
        - [ ] On <date 6 months from now>, 6 months from now, the component will be removed
```

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 2]
```diff
@@ -0,0 +1,290 @@
+name: Unmaintained component
+description: Create an issue to track the deprecation of a component
+labels: ["help wanted", "unmaintained"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report! Please make sure to fill out the entire form below, providing as much context as you can in order to help us triage and track down your bug as quickly as possible.
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml]: I don't have a great suggestion here, but maybe change the wording in the second half of this statement so it's not referring to this type of issue as a bug? (Same comment for the next line here as well)

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+        Before filing a bug, please be sure you have searched through [existing bugs](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug) to see if an existing issue covers your bug.
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml]:
```suggestion
        Before filing a bug, please search through [existing bugs](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug) to see if an existing issue covers your bug.
```
Nit, feel free to drop.

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+  - type: dropdown
+    id: component
+    attributes:
+      label: Component(s)
+      description: Which component no longer maintained?
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml:14]:
```suggestion
      description: Which component is no longer maintained?
```

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 5]
```diff
@@ -0,0 +1,289 @@
+name: Unmaintained component
+description: Create an issue to track the deprecation of a component
+labels: ["help wanted", "unmaintained"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please open an issue using this template if code owners for a component have been unresponsive in 6 weeks. Ensure
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml:8]:
```suggestion
        Please open an issue using this template if code owners for a component have been unresponsive for at least 6 weeks. Ensure
```
Nit: feel free to drop if this isn't more clear than what's there already.

File: .github/ISSUE_TEMPLATE/unmaintained.yaml [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+        links to relevant pull requests and issues are added to this issue.
```
Comment (crobert-1) [inline .github/ISSUE_TEMPLATE/unmaintained.yaml:9]: Would it make sense to add a specific section for relevant links?

Comment (codeboten) [reply .github/ISSUE_TEMPLATE/unmaintained.yaml:9]: Added a section at the bottom of the template

File: .github/ISSUE_TEMPLATE/unmaintained.yaml (hunk 3 above)
Comment (codeboten) [reply .github/ISSUE_TEMPLATE/unmaintained.yaml]: Updated this entire section to make it more relevant

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**